bson/bsoncodec/codec_cache_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsoncodec
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+var codecCacheTestTypes = [16]reflect.Type{
```

Comment:
Yep and I defined it as 16 to be explicit that the number of elements is a power of 2, which is important because we use mod `len(codecCacheTestTypes)` in the benchmarks and I don't want the modular arithmetic (even though it's a const op) to be a factor in the bench results.

Reply:
Thanks for the explanation! Consider adding a comment describing your reasoning, but otherwise consider this comment resolved.